src/NUnitFramework/tests/Internal/TestWorkerTests.cs

```diff
@@ -56,15 +56,20 @@ public void BusyExecuteIdleEventsCalledInSequence()
             StringBuilder sb = new StringBuilder();
             FakeWorkItem work = Fakes.GetWorkItem(this, "FakeMethod");
 
-            _worker.Busy += (s, ea) => { sb.Append("Busy"); };
-            work.Executed += (s, ea) => { sb.Append("Exec"); };
-			_worker.Idle += (s, ea) => { sb.Append ("Idle"); };
+            using (var countdown = new CountdownEvent(3))
+            {
+                _worker.Busy += (s, ea) => { sb.Append("Busy"); countdown.Signal(); };
+                work.Executed += (s, ea) => { sb.Append("Exec"); countdown.Signal(); };
+                _worker.Idle += (s, ea) => { sb.Append("Idle"); countdown.Signal(); };
 
-            _queue.Enqueue(work);
-            _worker.Start();
-            _queue.Start();
+                _queue.Enqueue(work);
+                _worker.Start();
+                _queue.Start();
 
-            Assert.That(() => sb.ToString(), Is.EqualTo("BusyExecIdle").After(200));
+                if (!countdown.Wait(10000)) Assert.Fail("Timed out.");
```

Comment:
If I had just changed `After`, we would have an unnecessary 10 second wait on success. (I also avoid [Timeout] on tests when possible because thread aborts introduce inherent unreliability.)

Reply:
Would this work using the polling functionality of DelayedConstraint? (E.g. After(10000,200))

Reply:
I wasn't aware of that. Probably! Should  I try it?

Reply:
Seems to work great, thanks!